R/future_promise.R

```diff
@@ -176,7 +177,26 @@ WorkQueue <- R6::R6Class("WorkQueue",
 
     # Returns a logical which let's work begin
     can_proceed = function() {
-      isTRUE(private$can_proceed_fn())
+      can_proceed_cache_val <- private$can_proceed_cache_val
+      # Return early if no work can be submitted.
+      if (!is.null(can_proceed_cache_val)) return(can_proceed_cache_val)
+
+      ret <- isTRUE(private$can_proceed_fn())
+
+      # If is `FALSE`, then any other attempts to start work in this tick will also fail.
```

Comment:
I would use some other terminology than "tick", since that doesn't really make sense for later. (It does make sense in a browser/JS context, but later is a bit different.)

Reply:
Maybe something roughly like this:

```suggestion
      # If is `FALSE`, then any other attempts to start work while looping over this set of future_promises will probably also fail.
```

Reply:
```suggestion
      # If is `FALSE`, then any other attempts within the current {later} loop execution should also fail
```

Reply:
Resolved in other comment. Closing